python/interpret_text/common/base_explainer.py

```diff
@@ -0,0 +1,29 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Defines the base text explainer API to create explanations."""
+from abc import abstractmethod
+from typing import Optional, Any, Callable, Iterable
+
+from interpret_text.common.base_text_model import BaseTextModel
+from interpret_text.common.preprocessor.base_text_preprocessor import BaseTextPreprocessor
+
+
+class BaseTextExplainer:
+
+    @abstractmethod
+    def explain_local(self, text: str, **kwargs) -> Callable:
+        raise NotImplementedError
```

Comment:
maybe use pass instead of NotImplementedError?

Reply:
@imatiach-msft - But shouldn't we enforce explain_local implementation when a new an explainer is created? pass will let the developer skip it?

Reply:
great point, I remember having this discussion with Walter and I had the exact same comment - but the idea is that the abstractmethod/abstract class will throw regardless if it's not implemented.  However, the way we define classes in the old way to be backwards compatible with older python versions, the ABC class does not throw for unimplemented methods.  I'm not sure if you are using the new or old way to do it.

Reply:
hmm it looks like you may be missing the extra step to define BaseTextExplainer as an abstract class

Reply:
Whatever the implementation here, it should match similar structures in interpret-community